crates/ruff_server/src/session/index.rs

```diff
@@ -264,31 +264,43 @@ impl Index {
         Some(controller.make_ref(cell_url, url, document_settings))
     }
 
-    /// Reloads relevant existing settings files based on a changed settings file path.
-    pub(super) fn reload_settings(&mut self, changed_url: &Url) {
-        let Ok(changed_path) = changed_url.to_file_path() else {
-            // Files that don't map to a path can't be a workspace configuration file.
-            return;
-        };
+    /// Reload the settings for all the workspace folders that contain the changed files.
+    ///
+    /// This method avoids re-indexing the same workspace multiple times if multiple files
+    /// belonging to the same workspace have been changed.
+    pub(super) fn reload_settings(&mut self, changes: &[FileEvent]) {
```

Comment:
I as a little confused at first why we aren't matching on the file names. It might be worth mentioning in the documentation that it's expected that the `FileEvent` are only for configuration files.